kroxylicious-sample/src/test/java/io/kroxylicious/sample/SampleFilterIntegrationTest.java

```diff
@@ -0,0 +1,154 @@
+    private static final String FIND_CONFIG_FIELD = "findValue";
+    private static final String REPLACE_CONFIG_FIELD = "replaceValue";
+    private static final Map<String, Object> PRODUCE_CONFIG = Map.of(FIND_CONFIG_FIELD, "foo", REPLACE_CONFIG_FIELD, "bar");
+    private static final Map<String, Object> FETCH_CONFIG = Map.of(FIND_CONFIG_FIELD, "bar", REPLACE_CONFIG_FIELD, "baz");
+    private static final Integer TIMEOUT_SECONDS = 10;
+    private static final String TOPIC_NAME = "test";
+    private static final Integer TOPIC_PARTITIONS = 1;
+    private static final Short TOPIC_REPLICATION = 1;
+
+    KafkaCluster cluster;
+    private KroxyliciousTester tester;
+
+    @BeforeEach
+    public void beforeEach() {
+        tester = kroxyliciousTester(withDefaultFilters(proxy(cluster))
+                .addToFilters(new FilterDefinitionBuilder(SampleContributor.SAMPLE_PRODUCE).withConfig(PRODUCE_CONFIG).build())
+                .addToFilters(new FilterDefinitionBuilder(SampleContributor.SAMPLE_FETCH).withConfig(FETCH_CONFIG).build()));
+    }
+
+    @AfterEach
+    public void afterEach() {
+        tester.close();
+    }
+
+    @Test
+    public void sampleFilterWillTransformRoundTripTest(Admin admin) {
+        try (Producer<String, String> producer = tester.producer();
+                Consumer<String, byte[]> kafkaClusterConsumer = new KafkaConsumer<>(mergeMaps(cluster.getKafkaClientConfiguration(),
+                        Map.of(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class, ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
+                                ByteArrayDeserializer.class, ConsumerConfig.GROUP_ID_CONFIG, "group-id-0", ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")));
+                Consumer<String, byte[]> proxyConsumer = tester.consumer(Serdes.String(), Serdes.ByteArray(),
+                        Map.of(ConsumerConfig.GROUP_ID_CONFIG, "group-id-1", ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest"))) {
+            admin.createTopics(List.of(new NewTopic(TOPIC_NAME, TOPIC_PARTITIONS, TOPIC_REPLICATION))).all().get(TIMEOUT_SECONDS, TimeUnit.SECONDS);
+            producer.send(new ProducerRecord<>(TOPIC_NAME, PRE_TRANSFORM_VALUE)).get(TIMEOUT_SECONDS, TimeUnit.SECONDS);
+            // get from cluster
+            kafkaClusterConsumer.subscribe(List.of(TOPIC_NAME));
+            ConsumerRecords<String, byte[]> kafkaClusterPoll = kafkaClusterConsumer.poll(Duration.ofSeconds(TIMEOUT_SECONDS));
+            if (kafkaClusterPoll.count() != 1) {
+                fail(String.format("Sent 1 record but received %d records from Kafka Cluster Consumer.", kafkaClusterPoll.count()));
+            }
+            ConsumerRecord<String, byte[]> clusterRecord = kafkaClusterPoll.records(TOPIC_NAME).iterator().next();
+            // check cluster record value is correct
+            assertEquals(PRODUCE_TRANSFORM_VALUE, new String(clusterRecord.value(), StandardCharsets.UTF_8));
```

Comment:
Personally I'd extract this as
```suggestion
            assertConsumerRecordHasValue(clusterRecord, PRODUCE_TRANSFORM_VALUE);
```

Reply:
Done!